ocaml/idl/datamodel.ml

```diff
@@ -6663,17 +6663,44 @@ let vdi_list_changed_blocks = call
     ~allowed_roles:_R_VM_OP
     ()
 
+module Vdi_nbd_server_info = struct
+  let vdi_nbd_server_info =
+    let lifecycle = [Published, rel_inverness, ""] in
+    create_obj
+      ~in_db:false
+      ~persist:PersistNothing
+      ~gen_constructor_destructor:false
+      ~lifecycle
+      ~in_oss_since:None
+      ~name:_vdi_nbd_server_info
+      ~descr:"Details for connecting to a VDI using the Network Block Device protocol"
+      ~gen_events:false
+      ~messages:[]
+      ~doccomments:[]
+      ~messages_default_allowed_roles:(Some []) (* No messages, so no roles allowed to use them *)
+      ~contents:
+      [ (* uid _vdi_nbd_server_info; The uuid is not needed here and only adds inconvenience. *)
+        field ~qualifier:DynamicRO ~lifecycle ~ty:String "exportname" "The exportname to request over NBD. This holds details including an authentication token, so it must be protected appropriately. Clients should regard the exportname as an opaque string or token.";
+        field ~qualifier:DynamicRO ~lifecycle ~ty:String "address" "An address on which the server can be reached; this can be IPv4, IPv6, or a DNS name.";
+        field ~qualifier:DynamicRO ~lifecycle ~ty:Int "port" "The TCP port";
+        field ~qualifier:DynamicRO ~lifecycle ~ty:String "cert" "The TLS certificate of the server";
+        field ~qualifier:DynamicRO ~lifecycle ~ty:String "subject" "For convenience, this redundant field holds a subject of the certificate.";
```

Comment:
Maybe some more explanation could be added here, like how users should use this field, e.g. the certificate should be valid for this hostname. (Unless they are expected to know it.)